File: src/main/java/com/sunuvtc/sunuvtcbackend/services/CourseService.java
package com.sunuvtc.sunuvtcbackend.services;


import com.sunuvtc.sunuvtcbackend.dtos.CourseDTO;

import java.time.LocalDateTime;
import java.util.List;

public interface CourseService {
    List<CourseDTO> getAllCourses();
    CourseDTO getCourseById(Long id);
    CourseDTO createCourse(CourseDTO dto);
    CourseDTO updateCourse(Long id, CourseDTO dto);
    void deleteCourse(Long id);
    List<CourseDTO> getCoursesByChauffeur(Long chauffeurId);
    List<CourseDTO> getCoursesByStatut(String statut);
    List<CourseDTO> getCoursesByDateRange(LocalDateTime start, LocalDateTime end);
}

File: frontend/src/app/services/course.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Course } from '../models/course.model';

@Injectable({ providedIn: 'root' })
export class CourseService {
    private apiUrl = 'http://localhost:8080/api/courses';

    constructor(private http: HttpClient) {}

    getAllCourses(): Observable<Course[]> {
        return this.http.get<Course[]>(this.apiUrl);
    }

    getCourseById(id: number): Observable<Course> {
        return this.http.get<Course>(`${this.apiUrl}/${id}`);
    }

    getCoursesByChauffeur(chauffeurId: number): Observable<Course[]> {
        return this.http.get<Course[]>(`${this.apiUrl}/by-chauffeur/${chauffeurId}`);
    }

    getCoursesByStatut(statut: string): Observable<Course[]> {
        return this.http.get<Course[]>(`${this.apiUrl}/by-statut/${statut}`);
    }

    getCoursesByDateRange(start: Date, end: Date): Observable<Course[]> {
        const params = { start: start.toISOString(), end: end.toISOString() };
        return this.http.get<Course[]>(`${this.apiUrl}/by-date-range`, { params });
    }

    createCourse(course: Course): Observable<Course> {
        return this.http.post<Course>(this.apiUrl, course);
    }

    updateCourse(id: number, course: Course): Observable<Course> {
        return this.http.put<Course>(`${this.apiUrl}/${id}`, course);
    }

    deleteCourse(id: number): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/${id}`);
    }


}
